src/redux/store.ts

```diff
@@ -0,0 +1,57 @@
+import { configureStore } from "@reduxjs/toolkit";
+import { PayloadAction } from "@reduxjs/toolkit";
+import { Product } from "../types/Product";
+
+export interface CartItem {
+    id:string;
+    product:Product
+}
+
+interface CartState {
+    count: number;
+    items: Array<CartItem>;
+}
+
+const initialState: CartState = {
+    count: 0,
+    items: []
+}
+
+
+const cartReducer = (state: CartState = initialState, action: PayloadAction<CartItem>) => {
+    switch (action.type){
+        case "add":
+        {
+            const count = state.count + 1;
+
+            const products = [...state.items, action.payload];
+            return {count: count, items: products}
```

Comment:
the value of count is the same as `products.length`. No need to manually increment or decrement it.